doc/source/_static/js/whatsnew_links.js

```diff
@@ -0,0 +1,32 @@
+(function() {
+	const links = document.getElementsByTagName('a');
+	for (let i = 0; i < links.length; i++) {
+		const re = /(whatsnew.html)#(whatsnew)-[\d]+(-\w+)+/g;
+		let linkElements = links[i].href.split("/");		
+		if (re.test(linkElements.slice(-1)[0])) {
+			let whatsNew = linkElements.slice(-1)[0].split("#");
+			whatsNew = generateWhatsNew(whatsNew);
+			linkElements[linkElements.length - 1] = whatsNew;
+			links[i].href = linkElements.join("/")
+		}
+	}
+	/* Reformats last part of 'whatsnew' URLs */
+	function generateWhatsNew(whatsNew) {
+		let url = whatsNew[0];
+		let anchor = whatsNew[1];
+		url = replaceURL(url);
+		let anchorEls = anchor.slice(9).split("-");
+		anchorEls[0] = reformatVersion(anchorEls[0]);
+		return url + anchorEls[0] + anchorEls.slice(1).join("-");
+	}
+
+    /* Replaces 'whatsnew.html' with 'whatsnew/' */
+	function replaceURL(url) {
+		return url.split(".")[0] + "/"
+	}
+
+    /* Converts version number into vX.X.X */
+	function reformatVersion(version) {
+		return "v" + version.slice(0,1) + "." + version.slice(1,-1) + "." + version.slice(-1) + ".html#"
+	}
+})();
```

Comment:
Newline at end of file.

Reply:
Also, are these `whatsnew` functions parameterized on the version of the `whatsnew` that we are parsing?

Reply:
So the function assumes the format according to all the other links (https://github.com/pandas-dev/pandas/issues/23695) and assumes the version number is vX.X(X).X; so 1 digit, 1 or 2 digits and then 1 digit. Do you think it should be more general?

Reply:
Ah, sorry, so this wasn't 100% clear.  When I said "parameterized on version," I was referring to this comment in the original issue:

> We should be able to detect this in Javascript. When we're on the whatsnew for 0.24.0 or greater, we want to look for anchors pointing to /whatsnew.html#whatsnew-<version> for 0.23.4 or older. Then we'd rewrite /whatsnew.html#whatsnew-<version>-<title> to /whatsnew/<version>.html#<title>.

So this is referring to your script in general.  I didn't see any checks for version?

Reply:
Ahh OK I missed this part, I shall amend this!

Reply:
I have a fix to check for the version number, but since I use a regex to check if the link format is '#whatsnew-XXX-...' before attempting to change the URL, this should prevent this issue, since all links with v0.23.5 or greater will be of the new format of 'v.0.28.9-enhancements...' and so would fail the regex match anyways? 

If that's not the case, I'll push this latest fix to check for version number and only amend links that are v0.23.4 or older.